services/models.py

```diff
@@ -138,6 +145,74 @@ class ServiceClientId(models.Model):
     client_id = models.CharField(max_length=256, null=False, blank=False, unique=True)
 
 
+def _validate_gdpr_api_errors(errors):
+    try:
+        iter(errors)
+    except TypeError:
+        return False
+
+    expected_keys = {"code", "message"}
+    for error in errors:
+        if set(error.keys()) != expected_keys:
+            return False
+        if not error.get("code") or not isinstance(error["code"], str):
+            return False
+        if not error.get("message") or not isinstance(error["message"], dict):
+            return False
+
+        for key, value in error["message"].items():
+            if not key or not isinstance(value, str):
+                return False
+
+    return True
+
+
+def _format_gdpr_api_errors(errors):
+    """Returns new dictionary containing errors formatted to conform ServiceConnectionDeletionError
```

Comment:
Formatting output according to GraphQL schema doesn't sound like something that should be done in a `models.py` file. Any transformation to GraphQL schema should happen where the GraphQL output is needed.

Reply:
I'd like to see the errors from the client and errors created by profile BE in some internal format. Maybe I'll change the comment to just note that this is the internal format.